### PR TITLE
Spell check of Flutter docs

### PR DESCRIPTION
**Pull request by gspencergoog**
No code changes, just comments: I spell-checked all the comments in the repo.

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -32,7 +32,7 @@ final String pubCache = path.join(flutterRoot, '.pub-cache');
 final String toolRoot = path.join(flutterRoot, 'packages', 'flutter_tools');
 
 /// The arguments to pass to `flutter test` (typically the local engine
-/// configuration) -- prefilled with the arguments passed to test.dart.
+/// configuration) -- pre-filled with the arguments passed to test.dart.
```
Comment (natebosch) [inline dev/bots/test.dart]: "prefill" is more common than "pre-fill"

https://books.google.com/ngrams/graph?content=prefill%2C+pre-fill&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cprefill%3B%2Cc0%3B.t1%3B%2Cpre%20-%20fill%3B%2Cc0#t1%3B%2Cprefill%3B%2Cc0%3B.t1%3B%2Cpre%20-%20fill%3B%2Cc0

File: packages/flutter/lib/src/painting/binding.dart [hunk 2]
```diff
@@ -106,7 +106,7 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   /// widgets when this happens.
   ///
   /// Objects that show text and/or measure text (e.g. via [TextPainter] or
-  /// [Paragraph]) should listen to this and redraw/remeasure.
+  /// [Paragraph]) should listen to this and redraw/re-measure.
```
Comment (natebosch) [inline packages/flutter/lib/src/painting/binding.dart]: "remeasure" is more common than "re-measure"

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 3]
```diff
@@ -83,7 +83,7 @@ class PlaceholderDimensions {
 ///
 /// See [Text.textWidthBasis], for example.
 enum TextWidthBasis {
-  /// Multiline text will take up the full width given by the parent. For single
+  /// Multi-line text will take up the full width given by the parent. For single
```
Comment (natebosch) [inline packages/flutter/lib/src/painting/text_painter.dart]: "multiline" is more common than "multi-line"

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 4]
```diff
@@ -178,7 +178,7 @@ class TextPainter {
   /// This and [textDirection] must be non-null before you call [layout].
   ///
   /// The [InlineSpan] this provides is in the form of a tree that may contain
-  /// multiple instances of [TextSpan]s and [WidgetSpan]s. To obtain a plaintext
+  /// multiple instances of [TextSpan]s and [WidgetSpan]s. To obtain a plain text
```
Comment (natebosch) [inline packages/flutter/lib/src/painting/text_painter.dart]: "plaintext" is (slightly) more common than "plain text"

https://books.google.com/ngrams/graph?content=plaintext%2C+plain+text&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cplaintext%3B%2Cc0%3B.t1%3B%2Cplain%20text%3B%2Cc0

File: packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart [hunk 5]
```diff
@@ -228,7 +228,7 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
   /// The integrity check consists of:
   ///
   /// 1. Verify that the children index in childList is in ascending order.
-  /// 2. Verify that there is no dangling keepalive child as the result of [move].
+  /// 2. Verify that there is no dangling keep-alive child as the result of [move].
```
Comment (natebosch) [inline packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart]: "keepalive" is more common than "keep-alive"

File: packages/flutter/lib/src/services/text_input.dart [hunk 6]
```diff
@@ -418,7 +418,7 @@ class TextInputConfiguration {
   ///
   /// This flag only affects Android. On iOS, suggestions are tied directly to
   /// [autocorrect], so that suggestions are only shown when [autocorrect] is
-  /// true. On Android autocorrection and suggestion are controlled separately.
+  /// true. On Android auto-correction and suggestion are controlled separately.
```
Comment (natebosch) [inline packages/flutter/lib/src/services/text_input.dart]: "autocorrect" is more common than "auto-correction"

**Review by natebosch — COMMENTED** [review]
I think this spell checker may be too zealous on compound words...

**Comment by gspencergoog** [comment]
Hmm. I much prefer all of the hyphenated ones, but maybe that just makes me uncommon. The non-hyphenated versions all look like made up words to me.

I reverted all the hyphenations I added, and all other hyphenations of these words in all the comments, so that we're at least consistent.

I kept `plain text` because it feels more descriptive and less a term of art (and it's nearly as common).

File: packages/flutter/lib/src/material/slider_theme.dart [hunk 7]
```diff
@@ -1500,7 +1500,7 @@ abstract class BaseSliderTrackShape {
 /// [parentBox]. The track rectangle extends to the bounds of the [parentBox],
 /// but is padded by the [RoundSliderOverlayShape] radius. The height is defined
 /// by the [SliderThemeData.trackHeight]. The color is determined by the
-/// [Slider]'s enabled state and the track segments's active state which are
+/// [Slider]'s enabled state and the track segments' active state which are
```
Comment (Hixie) [inline packages/flutter/lib/src/material/slider_theme.dart]: `segment's`?

File: packages/flutter/lib/src/material/slider_theme.dart [hunk 8]
```diff
@@ -1706,7 +1706,7 @@ class RoundedRectSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
 /// [parentBox]. The track rectangle extends to the bounds of the [parentBox],
 /// but is padded by the [RoundSliderOverlayShape] radius. The height is
 /// defined by the [SliderThemeData.trackHeight]. The color is determined by the
-/// [Slider]'s enabled state and the track segments's active state which are
+/// [Slider]'s enabled state and the track segments' active state which are
```
Comment (Hixie) [inline packages/flutter/lib/src/material/slider_theme.dart]: ditto

File: packages/flutter_goldens/lib/flutter_goldens.dart [hunk 9]
```diff
@@ -156,7 +156,7 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
     return goldenFile;
   }
 
-  /// Prepends the golden Uri with the library name that encloses the current
+  /// Prepends the golden URI with the library name that encloses the current
```
Comment (Hixie) [inline packages/flutter_goldens/lib/flutter_goldens.dart]: URL

You should ban URI from your dictionary

File: packages/flutter_tools/lib/src/asset.dart [hunk 10]
```diff
@@ -257,11 +257,11 @@ class _Asset {
 
   final String baseDir;
 
-  /// A platform-independent Uri where this asset can be found on disk on the
+  /// A platform-independent URI where this asset can be found on disk on the
   /// host system relative to [baseDir].
   final Uri relativeUri;
 
-  /// A platform-independent Uri representing the entry for the asset manifest.
+  /// A platform-independent URI representing the entry for the asset manifest.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/asset.dart]: ditto

File: packages/flutter_tools/lib/src/build_runner/web_fs.dart [hunk 11]
```diff
@@ -100,7 +100,7 @@ class WebFs {
     this._dartDefines,
   );
 
-  /// The server uri.
+  /// The server URI.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/build_runner/web_fs.dart]: ditto

File: packages/flutter_tools/lib/src/build_system/source.dart [hunk 12]
```diff
@@ -182,7 +182,7 @@ class SourceVisitor implements ResolvedFiles {
 
 /// A description of an input or output of a [Target].
 abstract class Source {
-  /// This source is a file-uri which contains some references to magic
+  /// This source is a file URI which contains some references to magic
   /// environment variables.
   const factory Source.pattern(String pattern, { bool optional }) = _PatternSource;
   /// The source is provided by an [Artifact].
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/build_system/source.dart]: ditto x3

File: packages/flutter_tools/lib/src/dart/package_map.dart [hunk 13]
```diff
@@ -46,7 +46,7 @@ class PackageMap {
   /// Returns the path to [packageUri].
   String pathForPackage(Uri packageUri) => uriForPackage(packageUri).path;
 
-  /// Returns the path to [packageUri] as Uri.
+  /// Returns the path to [packageUri] as URI.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/dart/package_map.dart]: ditto

File: packages/flutter_tools/lib/src/devfs.dart [hunk 14]
```diff
@@ -534,5 +534,5 @@ class DevFS {
   }
 }
 
-/// Converts a platform-specific file path to a platform-independent Uri path.
+/// Converts a platform-specific file path to a platform-independent URI path.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/devfs.dart]: ditto

File: packages/flutter_tools/lib/src/device.dart [hunk 15]
```diff
@@ -545,7 +545,7 @@ class DebuggingOptions {
   final int deviceVmServicePort;
   final String port;
   final String hostname;
-  /// A file where the vmservice uri should be written after the application is started.
+  /// A file where the vmservice URI should be written after the application is started.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/device.dart]: ditto

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/slider_theme.dart (hunk 7 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/slider_theme.dart]: Aaah, when I was doing my spell checking, I thought that the comment was referring to a plural number of segments' active state, not just one segment's active state.

Clearly not so.  Changed to `segment's`

File: packages/flutter/lib/src/material/slider_theme.dart (hunk 8 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/slider_theme.dart]: Fixed.

File: packages/flutter_goldens/lib/flutter_goldens.dart (hunk 9 above)
Comment (gspencergoog) [reply packages/flutter_goldens/lib/flutter_goldens.dart]: OK, done.  And replaced all (reasonable) instances of URI with URL in the other places where they appeared.

File: dev/devicelab/lib/framework/framework.dart [hunk 16]
```diff
@@ -143,7 +143,7 @@ class _TaskRunner {
     });
   }
 
-  /// Disables the keep-alive port, allowing the VM to exit.
+  /// Disables the keepalive port, allowing the VM to exit.
```
Comment (dramosv90) [inline dev/devicelab/lib/framework/framework.dart]:
```suggestion
  /// Disables the [keepAlivePort] allowing the VM to exit.
```

File: packages/flutter_tools/lib/src/protocol_discovery.dart [hunk 17]
```diff
@@ -63,20 +63,20 @@ class ProtocolDiscovery {
...
   Future<Uri> get uri {
     return uris.first;
   }
 
-  /// The discovered service URIs.
+  /// The discovered service URLs.
   ///
-  /// When a new observatory URI is available in [logReader],
-  /// the URIs are forwarded at most once every [throttleDuration].
+  /// When a new observatory UR: is available in [logReader],
```
Comment (dramosv90) [inline packages/flutter_tools/lib/src/protocol_discovery.dart]:
```suggestion
  /// When a new observatory URL: is available in [logReader],
```

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 18]
```diff
@@ -293,7 +293,7 @@ class FlutterPlatform extends PlatformPlugin {
 
   /// The test compiler produces dill files for each test main.
```
Comment (dramosv90) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: dill or dll?

**Review by darielopper — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/lib/src/test/flutter_platform.dart (hunk 18 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/test/flutter_platform.dart]: dill is the serialization format for kernel, a Dart intermediate language used by the tooling

File: dev/devicelab/lib/framework/framework.dart (hunk 16 above)
Comment (gspencergoog) [reply dev/devicelab/lib/framework/framework.dart]: That would be great, but it's a private member: `_keepAlivePort`, and there isn't any description to link to.

File: packages/flutter_tools/lib/src/protocol_discovery.dart (hunk 17 above)
Comment (gspencergoog) [reply packages/flutter_tools/lib/src/protocol_discovery.dart]: Whoops.  Thanks!

**Pull request merged**